Feature/scriban templates

bpedziwiatr opened the pull request:
Add support  for Scriban templates

bpedziwiatr-sii committed:
add scriban template support

bpedziwiatr-sii committed:
add support for Scriban template

bpedziwiatr-sii committed:
fix output dir

bpedziwiatr-sii committed:
add autogenerated mark in scriban templates

Kraviecc commented on DataverseGen.Cli/example-config.json:
```diff
@@ -2,6 +2,8 @@
+  "OutDirectory": "out-crm",
+  "TemplateName": "Main", // for T4 use dataversetemplate.tt
+  "TemplateEngine": "scriban",
+  "IsSingleOutputScriban": false
```
you could use a nested object in the config model to have separate place for different engines

Kraviecc commented on DataverseGen.Cli/Program.cs:
```diff
@@ -61,17 +62,29 @@ private static void Main(string[] args)
+                        break;
+
+                    case "t4":
+                        Generator gen2 = new Generator(config.TemplateName, config.OutDirectory, context);
```
choose more appropriate name than gen2

Kraviecc commented on DataverseGen.Cli/Program.cs:
```diff
@@ -61,17 +62,29 @@ private static void Main(string[] args)
                 Console.WriteLine("Start generator");
-                Generator gen2 = new Generator("dataversetemplate.tt", config.OutDirectory, context);
-                gen2.GenerateTemplate();
+                switch (config.TemplateEngine)
```
you could go for a common interface with "GenerateTemplate" as a contract. Then within the switch there would be simple assignments or even this could be extracted to factory

Kraviecc commented on DataverseGen.Core/Generators/Scriban/MultipleFileTemplate.cs:
```diff
@@ -0,0 +1,91 @@
+            _templateName = templateName;
+            switch (_templateName)
+            {
+                case "Main":
```
I think we should store configurations constants in one place ("Main" here, different engine types...)

Kraviecc commented on DataverseGen.Core/Generators/Scriban/MultipleFileTemplate.cs:
```diff
@@ -0,0 +1,91 @@
+            XrmContextTemplate= Template.Parse(File.ReadAllText(_xrmServiceContextTemplateFilePath));
+        }
+
+        private string _entityTemplateFilePath;
```
format/reorder file (?)

Kraviecc commented on DataverseGen.Core/Generators/Scriban/ScribanGenerator.cs:
```diff
@@ -0,0 +1,84 @@
+            if (_isSingleOutput)
+                SingleFile();
+            else
+            {
```
inconsistent braces policy (always with {} or no for single statements

bpedziwiatr-sii committed:
feature/scriban-templates:
fix: scriban template for fields empty string
fix: name in info generator for files output

bpedziwiatr-sii committed:
feature/scriban-tempplates:
add gitversion
update screen size
